http/server.lua

```diff
@@ -1323,7 +1320,9 @@ local exports = {
             helper  = set_helper,
             hook    = set_hook,
             url_for = url_for_httpd,
-            listen  = tcp_server_create,
+
+            -- Exposed to make it replaceable by a user.
+            tcp_server_f = socket.tcp_server,
```

Comment:
Reminder: discuss this API with @ligurio and either squash or discard.

Reply:
Discussed orally. I would prefer API suggested by you.

Reply:
Squashed commits.